tree/ntuple/v7/test/rfield_vector.cxx

```diff
@@ -1,51 +1,99 @@
 #include "ntuple_test.hxx"
 
-TEST(RNTuple, ClassVector)
+// A layer of indirection to hide std::vector's second template parameter.
+// This way we can generalize tests over RVec and std::vector using a template template parameter (see below).
+template <typename T>
+using Vector_t = std::vector<T>;
+
+template <template <typename> class Coll_t>
+void TestClassVector()
 {
    FileRaii fileGuard("test_ntuple_classvector.root");
 
    auto modelWrite = RNTupleModel::Create();
-   auto wrKlassVec = modelWrite->MakeField<std::vector<CustomStruct>>("klassVec");
+   auto wrKlassVec = modelWrite->MakeField<Coll_t<CustomStruct>>("klassVec");
    CustomStruct klass;
-   klass.a = 42.0;
-   klass.v1.emplace_back(1.0);
-   wrKlassVec->emplace_back(klass);
-   klass.a = 137.0;
-   klass.v1.emplace_back(2.0);
+   klass.a = 42.f;
+   klass.v1.emplace_back(1.f);
    wrKlassVec->emplace_back(klass);
 
    {
       RNTupleWriter ntuple(std::move(modelWrite),
-         std::make_unique<RPageSinkFile>("myNTuple", fileGuard.GetPath(), RNTupleWriteOptions()));
+                           std::make_unique<RPageSinkFile>("myNTuple", fileGuard.GetPath(), RNTupleWriteOptions()));
+      ntuple.Fill();
+
+      // enlarge
+      wrKlassVec->emplace_back(CustomStruct{1.f, {1.f, 2.f, 3.f}, {{42.f}}, "foo"});
+      wrKlassVec->emplace_back(CustomStruct{2.f, {4.f, 5.f, 6.f}, {{1.f}, {2.f, 3.f}}, "bar"});
+      ntuple.Fill();
+
+      // shrink
+      wrKlassVec->clear();
+      wrKlassVec->emplace_back(CustomStruct{3.f, {7.f, 8.f, 9.f}, {{4.f, 5.f}, {}}, "baz"});
       ntuple.Fill();
    }
 
    RNTupleReader ntuple(std::make_unique<RPageSourceFile>("myNTuple", fileGuard.GetPath(), RNTupleReadOptions()));
-   EXPECT_EQ(1U, ntuple.GetNEntries());
+   EXPECT_EQ(3U, ntuple.GetNEntries());
 
    auto viewKlassVec = ntuple.GetViewCollection("klassVec");
    auto viewKlass = viewKlassVec.GetView<CustomStruct>("_0");
    auto viewKlassA = viewKlassVec.GetView<float>("_0.a");
+   auto viewKlassV1 = viewKlassVec.GetView<std::vector<float>>("_0.v1");
```

Comment:
We could read one of the next three fields as `RVec` (to check reading `RVec` as `std::vector` and vice versa)

Reply:
uhm i'm not sure I understand the suggestion, could you propose a change? (there is another test for reading std::vectors as RVec, I don't know if we have tests for reading RVecs as std::vectors -- should I just change `std::vector<float>` to `ROOT::RVecF` at line 42 to exercise that path?)